File: 4948.cpp
#include <iostream>
using namespace std;

bool prime[123456 * 2 + 1];   

void isPrime()
{
	prime[1] = true;

	for (int i = 2; i <= 123456 * 2; i++) {
		if (prime[i])
			continue;
		for (long long j = (long long)i * i; j <= 123456 * 2; j += i) {
			int index = (int)j;
			prime[index] = true;
		}
	}
}

int main()
{
	isPrime();

	while (true) {
		int n;
		cin >> n;

		if (n == 0)
			break;

		int ret = 0;
		for (int i = n + 1; i <= 2 * n; i++) {
			if (!prime[i])
				ret++;
		}

		cout << ret << endl;
	}

	return 0;
}

File: 4948.py
prime = [True for i in range(246912+1)]

def isPrime():
    prime[1] = False
    prime[0] = False
    for i in range(2, 246912+1):
        if not prime[i]:
            continue
        for j in range(i*i, 246912+1, i):
            prime[j] = False

isPrime()
while True:
    n = int(input())
    if n == 0:
        break

    count = 0
    for i in range(n+1, 2*n+1):
        if prime[i]:
            count += 1

    print(count)
